src/mainloop.rkt

```diff
@@ -138,21 +138,31 @@
       #;(exp ,exp-x ,log-x)
       #;(log ,log-x ,exp-x))))
 
-(define (taylor-alt altn loc)
+(define (taylor-alt altn loc repr)
   (define expr (location-get loc (alt-program altn)))
   (define vars (free-variables expr))
+
+  ;; resugar/desugaring
+  (define prec (representation-name repr))
+  (define var-precs (for/list ([var vars]) (cons var prec)))
+
   (if (or (null? vars) ;; `approximate` cannot be called with a null vars list
-          (not (equal? (type-of expr (*var-reprs*)) 'real)))
+          (not (set-member? '(binary64 binary32) (type-of expr repr (*var-reprs*)))))
```

Comment:
This should refer to types, not representations. So the test isn't correct: posits should work.